Proteomics/ProteolyticDigestion/DigestionMotif.cs

```diff
@@ -120,39 +120,47 @@ public bool Fits(string sequence, int location)
             // check for inducing cleavage
             for (m = 0; m < InducingCleavage.Length && fits; m++) // handle patterns
             {
-                currentResidue = sequence[location + m];
-                if (!MotifMatches(InducingCleavage[m], currentResidue))
+                if (location + m >= sequence.Length)
```

Comment:
`for (m = 0; m < InducingCleavage.Length && fits; m++)`
change to 
`for (m = 0; m < InducingCleavage.Length && m < sequence.Length && fits; m++) `
(check my math)